backends/s3/s3.go

```diff
@@ -45,6 +45,14 @@ type Options struct {
 	AccessKey string `yaml:"access_key"`
 	SecretKey string `yaml:"secret_key"`
 
+	// Path to the field containing the access key,
+	// e.g. /etc/s3-secrets/access-key.
+	AccessKeyFilename string `json:"access_key_filename"`
```

Comment:
```suggestion
	AccessKeyFilename string `yaml:"access_key_filename"`
```

Reply:
Nit: in other places like [go-tlsconfig](https://github.com/PowerDNS/go-tlsconfig/blob/master/config.go#L24) and internal products we use `_file` as the suffix. A 'filename' also suggests that a full path is not accepted.